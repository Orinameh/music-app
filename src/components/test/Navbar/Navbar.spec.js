import React from 'react';
import { shallow } from 'enzyme';
import Navbar from '../../layout/Navbar';

describe('Navbar', () => {
	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<Navbar />)
	})

	it('should render correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should render a <nav />', () => {
		expect(wrapper.find('nav').length).toEqual(1)
	})

	it('should render a span', () => {
		expect(wrapper.find('span').length).toEqual(1)
	})
})


