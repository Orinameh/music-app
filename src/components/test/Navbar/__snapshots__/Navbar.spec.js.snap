// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navbar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <span
        className="navbar-brand mb-0 h1 mx-auto"
      >
        Lyric Finder
      </span>,
      "className": "navbar navbar-dark bg-dark mb-5",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Lyric Finder",
        "className": "navbar-brand mb-0 h1 mx-auto",
      },
      "ref": null,
      "rendered": "Lyric Finder",
      "type": "span",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <span
          className="navbar-brand mb-0 h1 mx-auto"
        >
          Lyric Finder
        </span>,
        "className": "navbar navbar-dark bg-dark mb-5",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lyric Finder",
          "className": "navbar-brand mb-0 h1 mx-auto",
        },
        "ref": null,
        "rendered": "Lyric Finder",
        "type": "span",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
